include/cinder/gl/GlslProg.h

```diff
@@ -45,11 +44,49 @@
 namespace cinder { namespace gl {
 	
 typedef std::shared_ptr<class GlslProg> GlslProgRef;
+	
+class UniformValueCache;
 
 class ShaderPreprocessor;
 
 class GlslProg : public std::enable_shared_from_this<GlslProg> {
   public:
+	
+	//! Resembles all the information Queryable of an attribute
+	struct Attribute {
```

Comment:
I can give accessors. I was thinking about a base class as well. Maybe GlslProgInfo that holds the redundant stuff?